### DISCUSSION TITLE
Galera vale a pena investir estudos em template engines como Razor, EJS, Handlebars, Twig??

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1689] [Perguntas e Respostas]
Galera, 
uma dúvida, qual a opinião de vocês quando a questão:
Vale a pena estudar e ir a fundo em templates engines como EJS/Handlebars no node, ou Razor no .Net Core? 

Digo porque atualmente todo projeto que começo, começo fazendo SPA com ReactJS e full orientado a serviços... Vocês já consideram as template engines como tecnologias defasadas? Ou ainda estão fortes no mercado e possuem seus cenários de destaque sobre SPAs?

Gostaria de opiniões, muito obrigado!

**Comment by ninetails** [comment]
Creio que os template engines atualmente só servem para mandar o HTML básico, talvez usado pelo que tiver no framework de back que for utilizado nativamente. Se for para fazer SPA, não acharia defasado mas encararia como algo secundário pros dias atuais, mesmo pq tb vc poderia fazer toda essa base no SSR usando o próprio React, quando necessário, usando os `renderToString`/`renderToNodeStream`. Ou mesmo um bom e simples _template string_.

Só abriria uma exceção pro Razor, que parece ser diferente no carinha novo chamado Blazor, que não cheguei a dar uma olhada mas roda o Razor com WASM (posso estar falando besteira pois não conheço muito sobre soluções Microsoft).

**Comment by eliseumds** [comment]
Pra renderizar páginas, penso sim que essas tecnologias estão ultrapassadas, mas ainda usa-se muito linguagens como Twig pra criar email templates no servidor, por ex.
